compiler/rustc_const_eval/src/interpret/projection.rs

```diff
@@ -301,6 +301,8 @@ where
                 self.project_constant_index(base, offset, min_length, from_end)?
             }
             Subslice { from, to, from_end } => self.project_subslice(base, from, to, from_end)?,
+            // We don't want anything happening here, this is here as a dummy.
+            Subtype(_) => base.transmute(base.layout(), self)?,
```

Comment:
can you put this next to `OpaqueCast`/into the same branch to make it clearer that these two behave in the same way?